### DISCUSSION TITLE
Bug when using storage locally with docker-compose

### DISCUSSION THREAD

**Discussion by nacho-carnicero** [supabase/supabase#9539] [General]
# Bug report

## Describe the bug

I've been trying to have Supabase running locally in my computer for testing purposes. I'm using the [docker-compose file](https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml) and I managed to have everything up and running. When I try to connect to the DB everything seems to be working fine, but when I try to use the storage module to upload/retrieve a picture I get the following error:

```
The file system does not support extended attributes or has the feature disabled
```
## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Use the [docker-compose file](https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml) to launch Supabase locally
2. Create a bucket in the storage DB (I did this with pgadmin4)
3. Try to upload a picture to the storage with the JS client:
```
  await client.storage.from(bucket).upload(key, blob, {
    contentType: blob.type,
    upsert: false,
    cacheControl: "public, max-age=31536000, immutable",
  });
```

## Expected behavior

The upload works without getting errors

## System information

- OS: macOS Big Sur (11.6)
- Browser: Chrome (95.0.4638.69) and Brave (1.31.87)
- Version of supabase-js: 1.21.0
- Version of Node.js: 14.15.4

**Comment by kiwicopple** [comment]
Hey @nacho-carnicero - thanks for reporting this. what are you using for the [storage backend](https://github.com/supabase/supabase/blob/63db9ebf27fc3360de507354c7f4ec52112b20ae/docker/docker-compose.yml#L97)? S3 or local disk?

**Comment by nacho-carnicero** [comment]
I am using the local disk.

**Comment by nacho-carnicero** [comment]
I found this in the [docker documention for MacOS](https://docker-docs.netlify.app/docker-for-mac/osxfs/#extended-attributes):

```
Extended attributes are not yet supported.
```

I guess that's the problem? Have you been able to run this on a Mac?

**Comment by nacho-carnicero** [comment]
Quick update here: I tried executing the same app in a Linux machine and the storage worked without problems, so it does sound like a MacOS specific issue (probably more a Docker for MacOS issue). Did you face this issue before as well?

**Comment by kiwicopple** [comment]
thanks for the update @nacho-carnicero - I have been swamped this week prepping for Launch Week. I'll get around to testing/fixing in the coming week unless someone else manages to jump in before me

Looks like there are similar errors experienced in other storage engines: https://github.com/osxfuse/osxfuse/issues/85

**Comment by soedirgo** [comment]
@nacho-carnicero yes this is a macOS-specific Docker limitation - we could document this somewhere. Right now you'd need to use S3-like backend for this to work.

**Comment by nacho-carnicero** [comment]
Ok thanks, for the answer, let me know if you want me to add a PR for documenting this

**Comment by globemediaofficial** [comment]
Any updates on this? I have to use the macOS version of docker because Debian Linux isn't working with supabase. Is there something I can do to the files to make them compatible? I'm getting the error `StorageException(message: Error, statusCode: 500, error: The file system does not support extended attributes or has the feature disabled.)` when uploading an image file from `Flutter`. I haven't added any "extended attributes".

**Comment by pckimlong** [comment]
I'm also getting this error with docker on Windows Server 2019

**Comment by burmecia** [comment]
[This PR](https://github.com/supabase/storage-api/pull/246) is to fix this issue on macOS. You will also need to choose `VirtioFS` to enable write file metadata attribute in Docker desktop -> Preferences -> General.

**Reply by ArwedSchmidt** [reply]
Did something broke this fix? Using VirtioFS, no matter which engine, yields the errror mentioned above, "The file system does not support extended attributes or has the feature disabled". 

Using s3-compose.yml makes this error fade, though another one arrives (no valid authentication) arrives for me which I cannot see a reason for (user token is correct, internal auth seems fine too)

**Comment by aantti** [comment]
It might the same thing as described here:
https://github.com/orgs/supabase/discussions/38608
https://github.com/orgs/supabase/discussions/38608#discussioncomment-14367046
